velox/functions/prestosql/window/tests/SimpleAggregatesTest.cpp

```diff
@@ -30,118 +30,96 @@ static const std::vector<std::string> kAggregateFunctions = {
+// window function tests. The parameters are based on the function being
+// tested and a specific over clause. The window function is tested for
+// the over clause and all combinations of frame clauses.
+struct WindowTestParam {
+  const std::string function;
+  const std::string overClause;
 };
 
-std::vector<TestParam> getTestParams(
-    const std::vector<std::string>& frameClauses) {
-  std::vector<TestParam> params;
-  for (auto aggregateFunction : kAggregateFunctions) {
-    for (auto frameClause : frameClauses) {
-      params.push_back({aggregateFunction, frameClause});
-    }
-  }
-  return params;
-}
-
-class SimpleAggregatesTest : public WindowTestBase {
+class WindowFunctionTest : public WindowTestBase {
  protected:
-  SimpleAggregatesTest(const TestParam& testParam)
-      : function_(testParam.aggregateFunction),
-        frameClause_(testParam.frameClause) {}
-
-  RowVectorPtr makeBasicVectors(vector_size_t size) {
-    return makeRowVector({
-        makeFlatVector<int32_t>(size, [](auto row) { return row % 10; }),
-        // These columns are used for k PRECEDING/FOLLOWING frame bounds. So
-        // they should have values >= 1.
-        makeFlatVector<int64_t>(size, [](auto row) { return row % 7 + 1; }),
-        makeFlatVector<int64_t>(size, [](auto row) { return row + 1; }),
-    });
-  }
-
-  RowVectorPtr makeSinglePartitionVector(vector_size_t size) {
-    return makeRowVector({
-        makeFlatVector<int32_t>(size, [](auto /* row */) { return 1; }),
-        // These columns are used for k PRECEDING/FOLLOWING frame bounds. So
-        // they should have values >= 1.
-        makeFlatVector<int64_t>(size, [](auto row) { return row % 50 + 1; }),
-        makeFlatVector<int64_t>(size, [](auto row) { return row + 1; }),
-    });
-  }
+  explicit WindowFunctionTest(const WindowTestParam& testParam)
+      : function_(testParam.function), overClause_(testParam.overClause) {}
 
-  void testWindowFunction(
-      const std::vector<RowVectorPtr>& vectors,
-      const std::vector<std::string>& overClauses) {
+  void testWindowFunction(const std::vector<RowVectorPtr>& vectors) {
     WindowTestBase::testWindowFunction(
-        vectors, function_, overClauses, frameClause_);
+        vectors, function_, {overClause_}, kFrameClauses);
   }
 
   const std::string function_;
-  const std::string frameClause_;
+  const std::string overClause_;
 };
 
-class MultiAggregatesTest : public SimpleAggregatesTest,
-                            public testing::WithParamInterface<TestParam> {
+std::vector<WindowTestParam> getWindowTestParams(
+    const std::vector<std::string>& functions,
+    const std::vector<std::string>& overClauses) {
+  std::vector<WindowTestParam> params;
+  for (auto function : functions) {
+    for (auto overClause : overClauses) {
+      params.push_back({function, overClause});
+    }
+  }
+  return params;
+}
+
+class AggregatesTest : public WindowFunctionTest,
```

Comment:
naming: test name should match file name

Reply:
Sorry missed that.